infrastructure/src/main/java/org/corfudb/infrastructure/BaseServer.java

```diff
@@ -29,40 +38,65 @@
 
     /** HandlerMethod for the base server. */
     @Getter
+    @Deprecated
     private final HandlerMethods handler = HandlerMethods.generateHandler(MethodHandles.lookup(), this);
 
+    /**
+     * RequestHandlerMethods for the Base server
+     */
+    @Getter
+    private final RequestHandlerMethods handlerMethods =
+            RequestHandlerMethods.generateHandler(MethodHandles.lookup(), this);
+
     @Override
+    @Deprecated
     public boolean isServerReadyToHandleMsg(CorfuMsg msg) {
         return getState() == ServerState.READY;
     }
 
+    @Override
+    public boolean isServerReadyToHandleReq(RequestMsg request) {
+        return getState() == ServerState.READY;
+    }
+
     public BaseServer(@Nonnull ServerContext context) {
         this.serverContext = context;
         executor = Executors.newFixedThreadPool(serverContext.getBaseServerThreadCount(),
                 new ServerThreadFactory("baseServer-", new ServerThreadFactory.ExceptionHandler()));
     }
 
     @Override
+    @Deprecated
```

Comment:
This code path shouldn't be used right? Can you just throw an unsupported exception instead of having a default implementation and removing it later? its less error-prone that way.

Reply:
It was used by KEEP_ALIVE, but I'll replace it by PING as suggested below and make the required changes in NettyClientRouter.